src/ui/ui_i18n/README.md

```diff
@@ -0,0 +1,325 @@
+    };
+  }
+
+  render() {
+    const {
+      name,
+      unreadCount,
+    } = this.state;
+
+    return (
+      <p>
+        <FormattedMessage
+          id="welcome"
+          defaultMessage={`Hello {name}, you have {unreadCount, number} {unreadCount, plural,
+            one {message}
+            other {messages}
+          }`}
+          values={{name: <b>{name}</b>, unreadCount}}
+          />
+          ...
+      </p>
+    );
+  }
+}
+```
+
+## Angular
+
+Angular wrapper has 4 entities: translation `provider`, `service`, `directive`
+and `filter`. Both the directive and the filter use the translation `service`
+with i18n engine under the hood.
+
+The translation `provider` is used for `service` configuration and
+has the following methods:
+- `addMessages(messages: Map<string, string>, [locale: string])` - provides a way to register
+translations with the library
+- `setLocale(locale: string)` - tells the library which language to use by given
+language key
+- `getLocale()` - returns the current locale
+- `setDefaultLocale(locale: string)` - tells the library which language to fallback
+when missing translations
+- `getDefaultLocale()` - returns the default locale
+- `defineFormats(formats: object)` - supply a set of options to the underlying formatter
+
+The translation `service` provides the only one method:
+- `translate(id: string, [{values: object, defaultMessage: string}])` – translate message by id
+
+The translation `filter` has the following syntax:
+```
+{{'translationId' | i18n[:{ values: object, defaultMessage: string }]}}
+```
+
+Where:
+- `translationId` - translation id to be translated
+- `values` - values to pass into translation
+- `defaultMessage` - will be used unless translation was successful
+
+The translation `directive` has the following syntax:
+```html
+<ANY
+  i18n-id="{string}"
+  [i18n-values="{object}"]
+  [i18n-default-message="{string}"]
+></ANY>
+```
+
+Where:
+- `i18n-id` - translation id to be translated
+- `i18n-values` - values to pass into translation
+- `i18n-default-message` - will be used unless translation was successful
+
+In order to initialize the translation service, we need to pass locale and
+localization messages from I18n engine into the `i18nProvider`:
+
+```js
+import { uiModules } from 'ui/modules';
+import i18n from 'i18n';
+
+uiModules.get('kibana').config(function (i18nProvider) {
+  i18nProvider.addMessages(i18n.getMessages());
+  i18nProvider.setLocale(i18n.getLocale());
+});
+```
+
+After that we can use i18n directive in Angular templates:
+```html
+<span
+  i18n-id="welcome"
+  i18n-default-message="Hello!"
+></span>
+```
+
+## Node.JS
+
+`Intl-messageformat` package assumes that the
+[Intl](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl)
+global object exists in the runtime. `Intl` is present in all modern
+browsers and Node.js 0.10+. In order to load i18n engine
+in Node.js we should simply `require()` this module (in Node.js, the
```

Comment:
__nit:__ we use `import` instead of `require` all over the place, so let's stick to that here too (and in example below).